core/src/main/java/org/apache/james/core/MailAddress.java

```diff
@@ -129,6 +131,28 @@ public static  MailAddress getMailSender(String sender) {
         }
     }
 
+    public static MailAddress asMailAddress(String address) {
+        try {
+            return new MailAddress(address);
+        } catch (AddressException e) {
+            throw new RuntimeException(e);
```

Comment:
I don't like that method : it's not clear it can throw (no documentation and no clue about what will happen). The rule is usually : throw runtime when there's not much to recover (think IOException, which unfortunalety is not runtime) and checked exception when the caller can do something about it. A problem during address parsing is something you can recover from (and asMailAddressOptional is here for that).
So I'm in favor of removing this method entirely (and BTW with the Try monad, we wouldn't have such problems)